src/util/ivmCache/index.js

```diff
@@ -0,0 +1,243 @@
+const { LRUCache } = require('lru-cache');
+const logger = require('../../logger');
+const stats = require('../stats');
+
+/**
+ * IVM Cache implementation using battle-tested lru-cache library
+ * Handles caching of IVM isolates with TTL support and proper cleanup
+ */
+class IvmCache {
+  constructor(options = {}) {
+    this.maxSize = parseInt(
+      options.maxSize !== undefined ? options.maxSize : process.env.IVM_CACHE_MAX_SIZE || '50',
+      10,
+    );
+    this.ttlMs = parseInt(
+      options.ttlMs !== undefined ? options.ttlMs : process.env.IVM_CACHE_TTL_MS || '1800000',
+      10,
+    ); // 30 min default
+
+    this.cache = new LRUCache({
```

Comment:
@cisse21 @ktgowtham  this might be ok for now if we only mirror for a customer that has less than `maxSize` transformations. before mirroring in environments with more than that we should have our own cache. I don't think we need an external library or anything. It should be fairly simple to build.

Every time we do a `get` we bump a counter. For the sake of this example let's say that `maxSize` is 10. Then we allow only the top 10 from the counters to be cached. Once an ivm expires (e.g. 3mins?) only one in the top 10 can be cached and take the place of the expired one.